### ISSUE TITLE
[BUG] flash attention split-KV combine writes out of bounds and returns NaN/garbage for contiguous (batch, seqlen, heads, head_size) inputs

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5235]
## Summary

`flash_fwd_splitkv_combine_kernel` advances its output base pointer by `head_size` per row but walks the rows of its tile by `out.stride(-3)`. The two agree only when `out.stride(-3) == head_size` — which happens to hold on the SDPA route (ATen hands the op `transpose(1, 2)` views) and in the GQA-decode swap path, which is why the accuracy suite and SDPA look fine.

For a contiguous `(batch, seqlen, num_heads, head_size)` query — the layout the op's schema documents, and what a direct `torch.ops.aten._flash_attention_forward` call naturally passes — `out.stride(-3)` is `num_heads * head_size`. Every tile row after the first lands `(num_heads - 1) * head_size` elements too far: writes run past the end of the output allocation (`compute-sanitizer` reports invalid global writes at `flash_kernel.py:1176`), rows that should be written never are (the `torch.empty` output comes back as garbage or NaN), and the strays corrupt whatever the allocator placed nearby — in the runs below, the returned `softmax_lse`. The split-KV path is on by default and its heuristic selects exactly these decode/small-batch shapes. The same kernel body and stride binding are vendored in the `_kunlunxin`, `_sunrise` and `_tsingmicro` backends; I could only test the mainline NVIDIA path.

The split-KV path was added in #662.

## Reproduction

```python
import torch

import flag_gems

torch.manual_seed(0)


def run(name, B, Hq, Hkv, Sq, Skv, D):
    # Contiguous (batch, seqlen, num_heads, head_size) — the layout
    # aten::_flash_attention_forward documents for dense inputs.
    q = torch.randn(B, Sq, Hq, D, device="cuda", dtype=torch.float16)
    k = torch.randn(B, Skv, Hkv, D, device="cuda", dtype=torch.float16)
    v = torch.randn(B, Skv, Hkv, D, device="cuda", dtype=torch.float16)

    ref_out, ref_lse, *_ = torch.ops.aten._flash_attention_forward(
        q, k, v, None, None, Sq, Skv, 0.0, False, False
    )
    with flag_gems.use_gems():
        res_out, res_lse, *_ = torch.ops.aten._flash_attention_forward(
            q, k, v, None, None, Sq, Skv, 0.0, False, False
        )

    nans = torch.isnan(res_out).sum().item()
    print(
        f"{name}: out max|diff|={(res_out.float() - ref_out.float()).abs().max().item():.3g}  "
        f"NaNs={nans}/{res_out.numel()}  "
        f"lse max|diff|={(res_lse.float() - ref_lse.float()).abs().max().item():.3g}"
    )


run("MHA decode ", 1, 8, 8, 1, 4096, 64)
run("prefill    ", 1, 2, 2, 256, 8192, 64)
run("GQA decode ", 1, 32, 8, 1, 4096, 64)
```

```
MHA decode : out max|diff|=nan  NaNs=10/512  lse max|diff|=8.92
prefill    : out max|diff|=nan  NaNs=285/32768  lse max|diff|=9.72
GQA decode : out max|diff|=3.05e-05  NaNs=0/2048  lse max|diff|=0
```

GQA decode survives only because the `seqlenq_ngroups_swapped` path produces an output view whose seq stride coincides with `head_size`. Under `compute-sanitizer --tool memcheck` (caching allocator disabled) the MHA decode case reports 30 invalid global writes at `flash_fwd_splitkv_combine_kernel`, `flash_kernel.py:1176`.

## Environment

- FlagGems master (75cf3ea)
- NVIDIA H100 NVL and L40S
- torch 2.6.0+cu124, triton 3.2.0
